Client/Frontend/Browser/Playlist/Cells/PlaylistCell.swift

```diff
@@ -0,0 +1,208 @@
+        $0.axis = .horizontal
+        $0.alignment = .center
+        $0.spacing = 15.0
+    }
+    
+    private let infoStackView = UIStackView().then {
+        $0.axis = .vertical
+        $0.alignment = .top
+        $0.spacing = 5.0
+    }
+    
+    private let separator = UIView().then {
+        $0.backgroundColor = UIColor(white: 1.0, alpha: 0.15)
+    }
+    
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        
+        self.preservesSuperviewLayoutMargins = false
+        self.selectionStyle = .none
+        
+        contentView.addSubview(iconStackView)
+        contentView.addSubview(infoStackView)
+        iconStackView.addArrangedSubview(thumbnailView)
+        infoStackView.addArrangedSubview(titleLabel)
+        infoStackView.addArrangedSubview(detailLabel)
+        contentView.addSubview(separator)
+        
+        thumbnailView.snp.makeConstraints {
+            // Keeps a 94.0px width on iPhone-X as per design
+            $0.width.equalTo(iconStackView.snp.height).multipliedBy(1.46875)
+        }
+        
+        iconStackView.snp.makeConstraints {
+            $0.leading.equalToSuperview().offset(12.0)
+            $0.top.bottom.equalToSuperview().inset(8.0)
+        }
+        
+        infoStackView.snp.makeConstraints {
+            $0.left.equalTo(iconStackView.snp.right).offset(8.0)
+            $0.right.equalToSuperview().offset(-15.0)
+            $0.centerY.equalToSuperview()
+            $0.top.greaterThanOrEqualTo(iconStackView.snp.top)
+            $0.bottom.lessThanOrEqualTo(iconStackView.snp.bottom)
+        }
+        
+        separator.snp.makeConstraints {
+            $0.left.equalTo(titleLabel.snp.left)
+            $0.right.bottom.equalToSuperview()
+            $0.height.equalTo(1.0 / UIScreen.main.scale)
+        }
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func updateThumbnail() {
+        if let image = thumbnailView.image {
+            let boundsScale = thumbnailView.bounds.size.width / thumbnailView.bounds.size.height
+            let imageScale = image.size.width / image.size.height
+            var drawingRect = thumbnailView.bounds
+
+            if boundsScale > imageScale {
+                drawingRect.size.width =  drawingRect.size.height * imageScale
+                drawingRect.origin.x = (thumbnailView.bounds.size.width - drawingRect.size.width) / 2.0
+            } else {
+                drawingRect.size.height = drawingRect.size.width / imageScale
+                drawingRect.origin.y = (thumbnailView.bounds.size.height - drawingRect.size.height) / 2.0
+            }
+
+            let path = UIBezierPath(roundedRect: drawingRect, cornerRadius: 5.0)
+            thumbnailMaskView.path = path.cgPath
+            thumbnailView.layer.mask = thumbnailMaskView
+        }
+    }
```

Comment:
Can you explain to me what we're doing here with thumbnails and why we're doing this custom masking?

Reply:
It's not called anymore. Will be removed. Previously it was needed because design wanted images centered with a black background and aspect fit.